numpy/core/shape_base.py

```diff
@@ -442,22 +442,25 @@ def _assignment_recursion(result, arrays, list_ndim, start_indicies,
         return arrays.shape
 
 
+def _concatenate_shapes(shapes, axis):
```

Comment:
docstring should mention that `concatenate(arrs, axis).shape == _concatenate_shapes([a.shape for a in arrs], axis)`